perf.groovy

```diff
@@ -37,3 +55,61 @@ def projectFolder = Utilities.getFolderName(project) + '/' + Utilities.getFolder
         }
     }
 }
+
+// Create the Linux/OSX/CentOS coreclr test leg for debug and release and each scenario
+[true, false].each { isPR ->
+    ['Ubuntu'].each { os ->
+        def osGroup = getOSGroup(os)
+        def architecture = 'x64'
+        def configuration = 'Release'
+        def newJob = job(Utilities.getFullJobName(project, "perf_${osGroup}", isPR)) {
+            steps {
+                // Set up the copies
+                // Coreclr build containing the tests and mscorlib
+                shell('curl http://dotnet-ci.cloudapp.net/job/dotnet_coreclr/job/master/job/release_windows_nt/lastSuccessfulBuild/artifact/bin/tests/tests.zip -o tests.zip')
```

Comment:
I did follow that process, and make sure the script worked locally on my local Linux box, before testing it within Jenkins jobs. However, because Jenkins machine's configuration was quite different from local Linux machine, it still needed multiple rounds of debugging to make it work in automation. 

For example, this run required copying official coreclr/corefx/coreclr test bits to Jenkins Linux machine, unzipping them to the right location to be found by the perf run script. As Jenkins machine used its own root dir "/mnt/j/workspace/smile21prc_coreclr/coreclr-perf/perf_Linux/", it was very easy to delploy some of those bits to wrong dirs and hence I had to debug those steps carefully which was issues specific to Jenkins machines and not able to be debugged in local run.

Another typical example was, for some reason, on Jenkins machine, with exactly the same script code, ./corerun was not marked as accessible, hence I had to add code below to change its mode, which was also not reproable for local runs.
          +chmod u+x ./corerun

My point is, for simple Jenkins runs, debugging scripts locally might be almost good enough. But for complex runs like this, rounds of debugging are still needed in real Jenkins environment. To make that debugging experience more efficient, having some steps separate (debuggable on the fly rather than hiding in part of a script which requires being checked in) is helpful, according to my experience in accomplishing this task.